storeshare_api/app/controllers/users_controller.rb

```diff
@@ -1,15 +1,28 @@
 class UsersController < ApplicationController
   def show
     @user = User.find(params[:id])
-    @role = Subletter.find_by(user_id: @user.id) || Renter.find_by(user_id: @user.id)
+    
+    if !@user.nil? then
+      @renter_data = Renter.find_by(user_id: @user.id)
+      @subletter_data = Subletter.find_by(user_id: @user.id)
 
-    render json: { user: @user, role: @role }
+      render json: { user: @user, renter_data: @renter_data, subletter_data: @subletter_data }
+    else
+      render json: { user: nil, renter_data: nil, subletter_data: nil }
+    end
   end
 
   def show_by_email
     @user = User.find_by(email: params[:user][:email])
```

Comment:
in this case, does params work as a hash of hashes?

Reply:
yup! is indeed strange haha